clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -5214,4 +5214,29 @@ def SignBitOp : CIR_Op<"signbit", [Pure]> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// StdInitializerListOp
+//===----------------------------------------------------------------------===//
+
+def StdInitializerListOp : CIR_Op<"std_initializer_list.ctor"> {
+  let summary = "Initialize std::initializer_list";
+  let description = [{
+    The `std_initializer_list.ctor` operation will initialize
+    `std::initializer_list<T>` with given arguments list.
+
+    ```cpp
+    initializer_list<int> v{1,2,3}; // initialize v with 1, 2, 3
+    ```
+
+    The type of each argument should be the same as template parameter of
+    `std::initializer_list` (aka `T` in `std::initializer_list<T>`).
+  }];
+  let arguments = (ins StructPtr:$initList, Variadic<CIR_AnyType>:$args);
+  let assemblyFormat = [{
+    $initList `:` type($initList) `(` ($args^ `:` type($args))? `)` attr-dict
```

Comment:
In the short term, I think just moving the `initList` type to the end is good enough for me. In the long term, it'd be nice if we could omit the types of the args entirely (since they can be inferred from the initlist type), but I don't have a sense of how involved that'd be, so it doesn't need to be part of this diff. Using `{}` delimiters instead of `()` would be more initializer-list like, but that can also be confused with regions, so I'm not sure if it's actually better. The discussion in #541 looks super relevant, thanks for linking it.